src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -98,52 +82,72 @@ internal static ImmutableArray<IArgument> DeriveArguments(ImmutableArray<BoundEx
                     argumentIndex = argumentsToParameters.IndexOf(parameterIndex);
                 }
```

Comment:
Not part of this change but the assignment in the declaration of `argumentIndex` is unnecessary. Also consider using conditional so `argumentIndex` can be assigned in the declaration: ```int argumentIndex = argumentsToParameters.IsDefault ? parameterIndex : ...;```